gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagActionReminderScheduler.java

```diff
@@ -129,14 +129,18 @@ public void execute(JobExecutionContext context) {
 
   /**
    * Creates a key for the reminder job by concatenating all dagAction fields and the eventTime of the dagAction to
-   * allow reminders for actions associated with multiple flow executions within a deadline period (e.g. another
-   * flow execution may occur before a flow finish or job start deadline expires)
+   * allow reminders for distinct action requests of the same flow execution within a deadline period (e.g. multiple
+   * kill requests for the same flow execution). 
```

Comment:
when you say "deadline period", that's not related to job-start or flow-completion deadlines, is it?  do you mean "reminder period" (approx. the 'linger' setting)?

also, this is when multiple instances of the same action on the same flow execution originate more than 'epsilon' apart, right?  and it's applicable to only KILL and RESUME, correct?

Reply:
it should be deadline or reminder period and more likely the former will actually cause the issue bc the SLA deadline (ie: job complete within 2 hours) is stored in this scheduler and will conflict with another SLA deadline reminder (job complete within 2 hours) stored in the Scheduler. The same conflict can also happen with the reminder period. 

The latter is precisely correct, yes.